src/utils/index.ts

```diff
@@ -0,0 +1,105 @@
+import { Contract } from '@ethersproject/contracts'
+import { getAddress } from '@ethersproject/address'
+import { AddressZero } from '@ethersproject/constants'
+import { JsonRpcSigner, Web3Provider } from '@ethersproject/providers'
+import { BigNumber } from '@ethersproject/bignumber'
+import { abi as IUniswapV2Router02ABI } from '@uniswap/v2-periphery/build/IUniswapV2Router02.json'
+import { ChainId, JSBI, Percent, Token, CurrencyAmount, Currency, ETHER } from '@pancakeswap/sdk'
+import { ROUTER_ADDRESS } from '../config/constants'
+import { BASE_BSC_SCAN_URLS } from '../config'
+import { TokenAddressMap } from '../state/lists/hooks'
+
+// returns the checksummed address if the address is valid, otherwise returns false
+export function isAddress(value: any): string | false {
+  try {
+    return getAddress(value)
+  } catch {
+    return false
+  }
+}
+
+export function getBscScanLink(
+  data: string | number,
+  type: 'transaction' | 'token' | 'address' | 'block' | 'countdown',
+  chainId: ChainId = ChainId.MAINNET,
+): string {
+  switch (type) {
+    case 'transaction': {
+      return `${BASE_BSC_SCAN_URLS[chainId]}/tx/${data}`
+    }
+    case 'token': {
+      return `${BASE_BSC_SCAN_URLS[chainId]}/token/${data}`
+    }
+    case 'block': {
+      return `${BASE_BSC_SCAN_URLS[chainId]}/block/${data}`
+    }
+    case 'countdown': {
+      return `${BASE_BSC_SCAN_URLS[chainId]}/block/countdown/${data}`
+    }
+    default: {
+      return `${BASE_BSC_SCAN_URLS[chainId]}/address/${data}`
+    }
+  }
+}
+
+// shorten the checksummed version of the input address to have 0x + 4 characters at start and end
+export function shortenAddress(address: string, chars = 4): string {
```

Comment:
Effectively a duplicate of 

https://github.com/pancakeswap/pancake-frontend/blob/66a00e5af08e7203e375a1a0b81173e98ba1fd68/src/utils/truncateWalletAddress.ts#L5

?